src/ui/MLFacetManager/MLFacetManager.ts

```diff
@@ -0,0 +1,225 @@
+import { Component } from '../Base/Component';
+import { MLFacet } from '../MLFacet/MLFacet';
+import { InitializationEvents } from '../../events/InitializationEvents';
+import { QueryEvents, IQuerySuccessEventArgs } from '../../events/QueryEvents';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { exportGlobally } from '../../GlobalExports';
+import { find } from 'underscore';
+import { IFacetResponse } from '../../rest/Facet/FacetResponse';
+import { $$ } from '../../utils/Dom';
+import { Utils } from '../../utils/Utils';
+import { ComponentOptions } from '../Base/ComponentOptions';
+
+export interface IMLFacetManagerOptions {
+  enableReorder?: boolean;
+  onUpdate?: IMLFacetManagerOnUpdate;
+  compareFacets?: IMLFacetManagerCompareFacet;
+}
+
+export interface IMLFacetManagerOnUpdate {
+  (facet: MLFacet, index: number): void;
+}
+
+export interface IMLFacetManagerCompareFacet {
+  (facetA: MLFacet, facetB: MLFacet): number;
+}
+
+export class MLFacetManager extends Component {
+  static ID = 'MLFacetManager';
+  static doExport = () => exportGlobally({ MLFacetManager });
+
+  /**
+   * The options for the MLFacet
+   * @componentOptions
+   */
+  static options: IMLFacetManagerOptions = {
+    /**
+     * Whether to allow the reordering of the facets depending on the server's response.
+     *
+     * **Default:** `true`
+     */
+    enableReorder: ComponentOptions.buildBooleanOption({ defaultValue: true, section: 'Filtering' }),
+    /**
+     * Specifies the event handler function to execute when the facets are updated on the query response
+     * For each `MLFacet`, you can provide a callback function to execute after the facet is reordered.
+     *
+     * **Note:**
+     * > You cannot set this option directly in the component markup as an HTML attribute. You must either set it in the
+     * > [`init`]{@link init} call of your search interface (see
+     * > [Components - Passing Component Options in the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsintheinitCall)),
+     * > or before the `init` call, using the `options` top-level function (see
+     * > [Components - Passing Component Options Before the init Call](https://developers.coveo.com/x/PoGfAQ#Components-PassingComponentOptionsBeforetheinitCall)).
+     *
+     * **Example:**
```

Comment:
We could create an external article and link to it instead (to keep the reference lean).

Maybe just remind me to do that when you close the Jira and select "Requires public documentation" :)

Reply:
Good, want me to remove it here then?

Reply:
Yes, let's remove it